### ISSUE TITLE
[Python] Roundtripping with ``from_pandas().to_pandas()`` changes timestamp/timedelta resolution

### ISSUE THREAD

**Issue by phofl** [apache/arrow#34877]
### Describe the bug, including details regarding any error messages, version, and platform.

With pandas 2.0 we support more resolutions than nanosecond resolution. E.g. the following should preserve the reso:

```
import pandas as pd
import pyarrow as pa
df = DataFrame({"a": [np.datetime64("2022-01-01")]}, dtype="datetime64[s]")
result = pa.Table.from_pandas(df).to_pandas()
result.dtypes
```

```
a    datetime64[ns]
dtype: object
```

Expected:

```
a    datetime64[s]
dtype: object
```

Tried on the latest nightly build

### Component(s)

Python

**Comment by jorisvandenbossche** [comment]
Yes, this is known, historically we have hardcoded nanoseconds, which we can relax if pandas>=2 is installed.

See https://github.com/apache/arrow/issues/33321, going to close this as a duplicate

**Comment by jorisvandenbossche** [comment]
Duplicate of https://github.com/apache/arrow/issues/33321
